### PR TITLE
refactor(compiler): simplify AOT tests

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: packages/compiler/test/aot/test_util.ts [hunk 1]
```diff
@@ -466,3 +446,202 @@ function directoryExists(dirname: string, data: MockData | undefined): boolean {
   let result = find(dirname, data);
   return !!result && typeof result !== 'string';
 }
+
+export type MockDirectoryArray = {
```
Comment (chuckjaz) [inline packages/compiler/test/aot/test_util.ts]: consider  `MockFlatFileArray`

File: packages/compiler/test/aot/test_util.ts [hunk 2]
```diff
@@ -466,3 +446,202 @@ function directoryExists(dirname: string, data: MockData | undefined): boolean {
...
+  fileName: string,
+  content: string
+}[];
+
+export type ConvertableToMockDirArray =
```
Comment (chuckjaz) [inline packages/compiler/test/aot/test_util.ts]: Consider `MockData`

**Review by chuckjaz — APPROVED** [review]
Some name changes.

File: packages/compiler/test/aot/test_util.ts (hunk 1 above)
Comment (tbosch) [inline packages/compiler/test/aot/test_util.ts]: Done

File: packages/compiler/test/aot/test_util.ts (hunk 2 above)
Comment (tbosch) [inline packages/compiler/test/aot/test_util.ts]: Done

**Pull request merged**